src/fsharp/fsc.fs

```diff
@@ -2206,6 +2255,7 @@ type private OutputCollector() =
     do System.Console.SetError (outWriter false)
     member x.GetTextAndClear() = lock output (fun () -> let res = output.ToArray() in output.Clear(); res)
 
+#if FX_RESIDENT_COMPILER
```

Comment:
Since the resident compiler is not included on CoreCLR as yet, you can remove the running-on-mono parts below